common/src/main/java/org/apache/cassandra/diff/RetryStrategyFactory.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.cassandra.diff;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.cassandra.diff.JobConfiguration.RetryOptions;
+
+public class RetryStrategyFactory {
+    public final static String IMPLEMENTATION_KEY = "impl";
+    private final static Logger logger = LoggerFactory.getLogger(RetryStrategyFactory.class);
+    private final RetryOptions retryOptions;
+
+    public RetryStrategyFactory(RetryOptions retryOptions) {
+        this.retryOptions = retryOptions;
+    }
+
+    public RetryStrategy create() {
+        if (retryOptions != null) {
+            return create(retryOptions);
+        } else {
+            logger.info("Retry is disabled.");
+            return RetryStrategy.NoRetry.INSTANCE;
+        }
+    }
+
+    public static RetryStrategy create(Map<String, String> parameters) {
```

Comment:
feels slightly expensive doing this for every query - maybe we could make the strategy factory pluggable instead? Then just a factory.get() to get the actual strategy instance for every query?